pretyping/detyping.ml

```diff
@@ -935,10 +935,13 @@ let (f_subst_genarg, subst_genarg_hook) = Hook.make ()
 let rec subst_glob_constr subst = DAst.map (function
   | GRef (ref,u) as raw ->
     let ref',t = subst_global subst ref in
-    if ref' == ref then raw else
-      let env = Global.env () in
-      let evd = Evd.from_env env in
-      DAst.get (detype Now false Id.Set.empty env evd (EConstr.of_constr t))
+    if ref' == ref then raw else (match t with
+        | None -> GRef (ref', u)
+        | Some t ->
+          let env = Global.env () in
+          let evd = Evd.from_env env in
+          let t = t.Univ.univ_abstracted in (* XXX This seems dangerous *)
```

Comment:
However u is not an instance (it's a glob_instance ie a list of level names) so it's not easy to do.

Reply:
This code is basically for substituting the `glob_constr` occurring in LtacX code. At first view, in such code, the instance can only be made of global universes. Would using `Pretyping.interp_instance` be ok?

More upstream, it seems that Ltac code should detect universe errors earlier. For instance, the following leads to a wrong-instance-length runtime error:
```
Ltac f x := exact (id@{Type} x). 
```
and this would also break at the time of an `Inline` substitution of functor arguments.